tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.cs

```diff
@@ -282,107 +269,131 @@ private void EnrichTagsWithGitMetadata(List<string> tags)
             _gitMetadataAddedToRequestTags = true;
         }
 
-        private void ProcessResponse(GetRcmResponse response, IDictionary<string, Product> products)
+        private void ProcessResponse(GetRcmResponse response)
         {
             if (Log.IsEnabled(LogEventLevel.Debug))
             {
-                string description = response.TargetFiles.Count > 0 ?
-                                        "with the following paths: " + string.Join(",", response.TargetFiles.Select(t => t.Path)) :
-                                        "that is empty.";
+                var description = response.TargetFiles.Count > 0 ? "with the following paths: " + string.Join(",", response.TargetFiles.Select(t => t.Path)) : "that is empty.";
                 Log.Debug("Received Remote Configuration response {ResponseDescription}.", description);
             }
 
-            var actualConfigPath =
-                response
-                   .ClientConfigs
-                   .Select(RemoteConfigurationPath.FromPath)
-                   .ToDictionary(path => path.Path);
+            var configByProducts = new Dictionary<string, List<RemoteConfiguration>>();
+            var receivedPaths = new List<string>();
 
-            var changedConfigurationsByProduct = GetChangedConfigurations()
-               .GroupBy(config => config.Path.Product);
-
-            // copy products
-            foreach (var productGroup in changedConfigurationsByProduct)
+            // handle new configurations
+            foreach (var clientConfig in response.ClientConfigs)
             {
-                var product = products[productGroup.Key];
+                var remoteConfigurationPath = RemoteConfigurationPath.FromPath(clientConfig);
+                receivedPaths.Add(remoteConfigurationPath.Path);
+                var signed = response.Targets.Signed.Targets;
+                var targetFiles = (response.TargetFiles ?? Enumerable.Empty<RcmFile>()).ToDictionary(f => f.Path, f => f);
 
-                try
+                if (!signed.TryGetValue(remoteConfigurationPath.Path, out var signedTarget))
                 {
-                    var configurations = productGroup.ToList();
-                    product.AssignConfigs(configurations);
+                    ThrowHelper.ThrowException($"Missing config {remoteConfigurationPath.Path} in targets");
                 }
-                catch (Exception e)
+
+                if (!_subscriptionsProductKeys.Contains(remoteConfigurationPath.Product))
                 {
-                    Log.Warning(e, "Failed to apply remote configurations for product {Product}", product?.Name);
+                    ThrowHelper.ThrowException($"Received config {remoteConfigurationPath} for a product that was not requested");
```

Comment:
Should we really throw? Wouldn't logging and `continue;` be better?

Reply:
it was like that before so I didnt want to change it but you re right it 's a shame not applying other updates, so fixed in 549eec7407f84c36068859253400159a48889a36